packages/search/src/elasticsearch/utils.ts

```diff
@@ -325,7 +342,23 @@ export const createStatusHistory = async (
     return
   }
 
-  const user: IUserModelData = await getUser(body.updatedBy || '', authHeader)
+  let isSystem = false
+  if (hasScope({ Authorization: authHeader }, 'notification-api'))
+    isSystem = true
+
+  const user = !isSystem
+    ? await getUser(body.updatedBy || '', authHeader)
+    : await (async () => {
+        const payload = getTokenPayload(authHeader)
+        const system = await getSystem(
+          { systemId: payload.sub },
+          { Authorization: authHeader }
+        )
+        return system
+      })()
+
+  if (typeof user.name !== 'object') user.name = [user.name]
```

Comment:
Can you clarify this condition a little? Is this a difference between a system user and a regular user? Would 

```ts
const name = isSystemUser(user) ? ... : ...
```

be safer here than a mutation?

Reply:
For a system user, `user.name` is a string where for a regular user `user.name` is 

`name: {
    use: string
    family: string
    given: string[]
  }[]
 `
 on which we use a find operation later. So thought, i could wrap it for a system's `user.name` in an array. I've added a comment here. But maybe there could be another way for this.

Reply:
Don't need it really, user can actually be null for a system